src/org/labkey/test/components/ui/domainproperties/EntityTypeDesigner.java

```diff
@@ -230,6 +232,27 @@ public Optional<WebElement> optionalWarningAlert()
         return elementCache().optionalWarningAlert();
     }
 
+    public String getStoredAmountDisplayUnits()
+    {
+        return elementCache().storedAmountDisplayUnitsSelect.getValue();
+    }
+
+    public T setStoredAmountDisplayUnits(String displayUnits)
+    {
+        elementCache().storedAmountDisplayUnitsSelect.select(displayUnits);
+        return getThis();
+    }
```

Comment:
These aren't needed here they are already defined in FMSampleTypeDefinition, and are not visible outside of the app (i.e. in LKS).

Reply:
OK, I only did these because I didn't know how to use an FMSampleTypeDefinition to drive the UI (does it deal with the UI? I traced into the code a bit where it's used but assumed it deals with the API when I didn't see our UI wrappers showing evidence of being aware they exist)
If I'm wrong here I'll happily change this

Reply:
Fair point, and I was mistaken in referring to FM SampleTypeDefinition (my bad). I knew this had already been implemented before I just forgot where. Access to the stored amount fields are already present and are exposed through SMSampleTypeDefinition. They are there because they are only required for sample types in the App.

The methods you have provided here are not used and can be removed.

An example of how you would set the stored amount using the existing code is:
```
    @Test
    public void testSearchOnAlternateBarcode() throws Exception
    {
        String sampleTypeName = "test_search_on_alternate_barcodes";
        SMSampleTypeDefinition sampleTypeDefinition = new SMSampleTypeDefinition(sampleTypeName, InventoryMetricUnit.UNIT);
        sampleTypeDefinition.setFields(getTestFields());

        var dgen = SampleTypeAPIHelper.createEmptySampleType(getProjectName(), sampleTypeDefinition);
```

The sample type definition returned can be used to create a sample type using the UI or the API.